flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCTableSource.java

```diff
@@ -151,18 +152,19 @@ public static Builder builder() {
 	}
 
 	private JDBCInputFormat getInputFormat() {
-		final RowTypeInfo rowTypeInfo = (RowTypeInfo) fromDataTypeToLegacyInfo(producedDataType);
+		final RowType rowType = (RowType) producedDataType.getLogicalType();
 		JDBCInputFormat.JDBCInputFormatBuilder builder = JDBCInputFormat.buildJDBCInputFormat()
 				.setDrivername(options.getDriverName())
 				.setDBUrl(options.getDbURL())
-				.setRowTypeInfo(new RowTypeInfo(rowTypeInfo.getFieldTypes(), rowTypeInfo.getFieldNames()));
+				.setRowType(rowType);
 		options.getUsername().ifPresent(builder::setUsername);
 		options.getPassword().ifPresent(builder::setPassword);
 
 		if (readOptions.getFetchSize() != 0) {
 			builder.setFetchSize(readOptions.getFetchSize());
 		}
 
+		final RowTypeInfo rowTypeInfo = (RowTypeInfo) fromDataTypeToLegacyInfo(producedDataType);
 		final JDBCDialect dialect = options.getDialect();
```

Comment:
My point is that we can get the `JdbcConverter` from dialect here, and pass to `JdbcInputFormat`. So that, `JdbcInputFormat` don't need a `RowType`.